bwi_logging/src/bwi_logging/logger_node.py

```diff
@@ -70,17 +75,34 @@ def main(argv=None):
     logdir.chdir()                      # change to that directory
 
     # get the file prefix
-    prefix = rospy.get_param('~prefix', 'bwi')
+    prefix = rospy.get_param('~prefix', DEFAULT_PREFIX)
     rospy.loginfo('logging with prefix: ' + str(prefix))
 
     # this is the command we will issue:
     cmd = ['rosbag', 'record', '-o' + prefix, '-j']
     cmd.extend(topics.split())
     rospy.loginfo('running command: ' + str(cmd))
 
+    # run the rosbag command
+    status = subprocess.call(cmd)
+
+    rospy.loginfo('rosbag returned status: ' + str(status))
```

Comment:
That's my understanding. It returns when the subprocess terminates, as a result of ROS shutdown.